### ISSUE TITLE
phamtecinc RFQ add RFQ Edit

### ISSUE THREAD

**Issue by Draviddktechnolab** [Draviddktechnolab/phamtecinc#1]
phamtecinc RFQ add RFQ Edit RFQ page make merging_line  un_merging_line  line qty note proper fixd this issue

**Comment by Draviddktechnolab** [comment]
// ================= UN-MERGE =================
$(document).on("click", ".un_merging_line", function (event) {
    event.preventDefault();

    var $checked = $('.checked_lines_hidden:checked');

    if ($checked.length != 1) {
        alert("Please select exactly one merged row to unmerge.");
        return;
    }

    var $chk = $checked.first();

    // get the main (target) row id
    var mainRowId = $chk.closest('td').find("input[name='subline[]']").val();

    // merged IDs string from hidden or attr
    var mergedStr = $chk.attr('data-merging-line') 
        || $("tr.line_" + mainRowId + " .merging_hidden_line_" + mainRowId).val();

    if (!mergedStr) return;

    var ids = mergedStr.split(',').map(function (v) { return v.trim(); }).filter(Boolean);

    ids.forEach(function (id) {
        var $row = $("tr.line_" + id);
        $("tr.line_" + id).removeClass("is-merginged");

        // always show row back
        $row.show();
        $("tr.subinfo.line_" + id).show();

        $row.find(".is_merge_hidden_line_" + id).val(1);

        if (id == mainRowId) {
            // reset target row to itself
            $row.find(".merging_hidden_line_" + id).val(id);
            $row.find(".checked_lines_hidden")
                .attr("data-line", id)
                .attr("data-merging-line", id)
                .prop('checked', false);
            return;
        }

        // --- STEP 1: remove cloned elements from target ---
        var $clones = $("tr.line_" + mainRowId + " .main-all-img-box-" + mainRowId)
            .find("[data-merged-from='" + id + "']").detach();

        // --- STEP 2: restore their input names back ---
        $clones.each(function () {
            $(this).find("input[name^='full_file_name']").each(function () {
                var orig = $(this).data("orig-name"); // we stored original name during merge
                if (orig) {
                    $(this).attr("name", orig);
                } else {
                    // fallback: reconstruct
                    $(this).attr("name", "full_file_name[" + id + "][]");
                }
            });
            $(this).removeAttr("data-merged-from").removeClass("box-new-merging-line");
        });

        // --- STEP 3: append them back to their original row container ---
        // $("tr.line_" + id + " .main-all-img-box-" + id).append($clones);

        // --- STEP 4: remove merged filenames/exts from main row ---
        $("tr.line_" + mainRowId + " td:eq(2) p.merge_file_name[data-origin='" + id + "']").remove();
        $("tr.line_" + mainRowId + " td:eq(3) p[data-origin='" + id + "']").remove();

        // reset this row’s merge info
        $row.find(".merging_hidden_line_" + id).val(id);
        $row.find(".checked_lines_hidden")
            .attr("data-line", id)
            .attr("data-merging-line", id)
            .prop('checked', false);
    });

    // --- Clean up main row ---
    $("tr.line_" + mainRowId + " .merging_hidden_line_" + mainRowId).val(mainRowId);
    $("tr.line_" + mainRowId + " .checked_lines_hidden")
        .attr("data-line", mainRowId)
        .attr("data-merging-line", mainRowId)
        .prop('checked', false);

    // IMPORTANT: keep only original filename/ext in target row
    var $fileNames = $("tr.line_" + mainRowId + " td:eq(2) .merge_file_name");
    if ($fileNames.length > 1) {
        $fileNames.slice(1).remove();
    }

    var $exts = $("tr.line_" + mainRowId + " td:eq(3) p");
    if ($exts.length > 1) {
        $exts.slice(1).remove();
    }

    // clear checkboxes
    $('.checked_lines_hidden').prop('checked', false);

    updateLineNumbers();
});

	 function updateLineNumbers() {
        $(".table-qty-value").removeClass(function(index, className) {
            return (className.match(/table-qty-value-line-\d+/g) || []).join(" ");
        });
		$("#preview_pdf_tbl tbody.super_block tr.mainlines:visible").each(function (index) {
            let newindex = parseInt(index) + parseInt(1);
		  $(this).find("td a.simplified_pop").text(index + 1);
          $(this).find(".table-qty-value").addClass("table-qty-value-line-" + newindex);
		});
	  }

$(document).on("click", ".btn-plues-qty", function(event){
    event.preventDefault();

    let mainlines = $("#preview_pdf_tbl tbody.super_block tr.mainlines:visible").not(".is-merginged");
    let dataplusqty = $(this).attr("data-plus-qty");
    if(dataplusqty < 5){
        dataplusqty = parseInt(dataplusqty) + parseInt(1);
        $(".btn-plues-qty").attr("data-plus-qty", dataplusqty);
    }
    mainlines.each(function () {
        let line = $(this).attr("dataid");
        let tableQtynext = $(".table-qty-value-line-" + line).find(".main-qty-block").length + 1;
        if(tableQtynext <= 5){
            $(".table-qty-value-line-" + line).append(`<div class="input-group main-qty-block" style="margin: 10px 0px;"><input type="text" class="form-control allqtybluers" data-qty="${tableQtynext}" name="qty_${tableQtynext}[${line}][]" placeholder="Qty ${tableQtynext}"><div class="input-group-addon"><a href="javascript:void(0)" class="btn-qty-delete" main-line="${line}" qty-line="${tableQtynext}"><i class="fa fa-trash" aria-hidden="true"></i></a></div></div>`);
            let iloopcount = line;
        }
    });
});
